CMakeLists.txt

```diff
@@ -0,0 +1,83 @@
+# enable x.y version format
+cmake_policy(SET CMP0048 NEW)
+project(CommonLib VERSION 0.1)
+
+cmake_minimum_required(VERSION 3.14)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+set(CMAKE_BUILD_TYPE Debug)
+
+find_package(Threads REQUIRED)
+
+set(protobuf_MODULE_COMPATIBLE TRUE)
+find_package(Protobuf CONFIG REQUIRED)
+message(STATUS "Using protobuf ${Protobuf_VERSION}")
+
+set(_PROTOBUF_LIBPROTOBUF protobuf::libprotobuf)
+set(_REFLECTION gRPC::grpc++_reflection)
+set(_PROTOBUF_PROTOC $<TARGET_FILE:protobuf::protoc>)
+
+find_package(gRPC CONFIG REQUIRED)
```

Comment:
make this support both install grpc and submodule grpc.
https://github.com/grpc/grpc/blob/master/examples/cpp/helloworld/CMakeLists.txt

Reply:
ok